modules/manually-restoring-cluster-etcd-backup.adoc

```diff
@@ -0,0 +1,299 @@
+----
+$ sudo cp /home/core/assets/backup/<snapshot_yyyy-mm-dd_hhmmss>.db /var/lib/etcd
+----
+
+.. Start a new container which includes the `etcdctl` tool and use a plain `bash` entrypoint:
++
+[source,terminal]
+----
+$ sudo podman run --rm --authfile=/var/lib/kubelet/config.json -it --entrypoint="/bin/bash" -v /var/lib/etcd:/var/lib/etcd:z registry.redhat.io/openshift4/ose-etcd:latest
+----
+
+.. Check that the version of the `etcdctl` tool is the version of the `etcd` server where the backup was created:
++
+[source,terminal]
+----
+$ sudo etcdctl version
+----
+
+.. Run the command to regenerate the `etcd` database, using the correct values for the current host. 
++
+[source,terminal]
+----
+$ sudo ETCDCTL_API=3 /usr/bin/etcdctl snapshot restore /var/lib/etcd/<snapshot_yyyy-mm-dd_hhmmss>.db \
+  --name "<ETCD_NAME>" \
+  --initial-cluster="<ETCD_INITIAL_CLUSTER>" \
+  --initial-cluster-token "openshift-etcd-<UUID>" \
+  --initial-advertise-peer-urls "<ETCD_NODE_PEER_URL>" \
+  --data-dir="/var/lib/etcd/restore-<UUID>" \
+  --skip-hash-check=true
+----
++
+[NOTE]
+====
+The quotes are mandatory when regenerating the `etcd` database.
+====
+
+. Write down the values printed in the `added member` logs; for example:
++
+.Example output
+----
+2022-06-28T19:52:43Z    info    membership/cluster.go:421   added member    {"cluster-id": "c5996b7c11c30d6b", "local-member-id": "0", "added-peer-id": "56cd73b614699e7", "added-peer-peer-urls": ["https://10.0.91.5:2380"], "added-peer-is-learner": false}
+2022-06-28T19:52:43Z    info    membership/cluster.go:421   added member    {"cluster-id": "c5996b7c11c30d6b", "local-member-id": "0", "added-peer-id": "1f63d01b31bb9a9e", "added-peer-peer-urls": ["https://10.0.90.221:2380"], "added-peer-is-learner": false}
+2022-06-28T19:52:43Z    info    membership/cluster.go:421   added member    {"cluster-id": "c5996b7c11c30d6b", "local-member-id": "0", "added-peer-id": "fdc2725b3b70127c", "added-peer-peer-urls": ["https://10.0.94.214:2380"], "added-peer-is-learner": false}
+----
+.. Exit from the container.
+
+.. Repeat these steps on all the control plane hosts, checking that the values printed in the added member logs are the same.
+
+. Move the regenerated `etcd` database to the default location.
+
+.. Move the regenerated database (that is, the `member` folder created by the previous`etcdctl snapshot restore`) in the etcd default location `/var/lib/etcd`:
++
+[source,terminal]
+----
+$ sudo mv /var/lib/etcd/restore-<UUID>/member /var/lib/etcd
+----
+
+.. Restore the SELinux context for `/var/lib/etcd/member` folder on `/var/lib/etcd` directory:
++
+[source,terminal]
+----
+$ sudo restorecon -vR /var/lib/etcd/
+----
+
+.. Remove the leftover files and directories:
++
+[source,terminal]
+----
+$ sudo rm -rf /var/lib/etcd/restore-<UUID>
+----
++
+[source,terminal]
+----
+$ sudo rm /var/lib/etcd/<snapshot_yyyy-mm-dd_hhmmss>.db
+----
++
+In the end, `/var/lib/etcd` must contain only the folder `member`.
```

Comment:
This is important, otherwise: https://access.redhat.com/solutions/6958920

Reply:
Done.